eduardhogea/MLOps
# Working GUI with BentoML support

# Integrate ZeroShotClassifier with BentoML Service

## Description

### Changes Made
1. **Model Development (`zero_shot_transformer.py`):**
   - Updated the `ZeroShotClassifier` for compatibility with BentoML.
   - Added functions to save and load the model using BentoML's model management features.

2. **BentoML Service Integration (`service.py`):**
   - Created a new BentoML service to serve the `ZeroShotClassifier`.
   - Implemented an API endpoint to receive text and candidate labels, and return classification results.

3. **GUI Application Update (`gui.py`):**
   - Modified the GUI application to send requests to the BentoML service.
   - Enhanced result display to include top classification result and a text snippet.



### How to Test
1. Start the BentoML service: `bentoml serve service.py:svc`
2. Run the GUI application: `python gui.py`
3. Perform text classification and observe the displayed results.
